FWCore/Integration/test/TransformIntProducer.cc

```diff
@@ -0,0 +1,41 @@
+#include "FWCore/Framework/interface/global/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "DataFormats/TestObjects/interface/ToyProducts.h"
+
+namespace edmtest {
+  class TransformIntProducer : public edm::global::EDProducer<edm::Transformer> {
+  public:
+    TransformIntProducer(edm::ParameterSet const& iPSet)
+        : getToken_(consumes(iPSet.getParameter<edm::InputTag>("get"))),
+          offset_(iPSet.getParameter<unsigned int>("offset")),
+          transformOffset_(iPSet.getParameter<unsigned int>("transformOffset")) {
+      putToken_ = produces<IntProduct>();
+      registerTransform(
+          putToken_,
+          [offset = transformOffset_](auto const& iFrom) { return IntProduct(iFrom.value + offset); },
```

Comment:
Would it be feasible to add a check that with `--noTransform` option the transformation function will not run?